### PR TITLE
[ENG-1201] Pre cleanup refactor

### PR DESCRIPTION
**Pull request by rerolled**
This PR cleans up the existing code without changing any functionality.

The bulk of the changes are moving the response and error handlers to their own functions. This massively decouples the request sending and handling code, and it allows us to comfortably move forward with additions and changes.

All tests are passing locally.

### CODE REVIEW

File: lib/request.js [hunk 1]
```diff
@@ -2,209 +2,215 @@
...
+ *        resolveWithFullResponse: true
+ *    });
+ */
+
+module.exports = function request(rawOptions) {
+    return new Promise((resolve, reject) => {
+        const options = _parseOptions(rawOptions);
+        const protocol = options.uri.split(':')[0];
+        const reqOpts = Object.assign({}, options, {
+            agent: options.keepAlive === false ? null : agent[protocol],
```
Comment (rerolled) [inline lib/request.js:58]: Moved request options `agent` reference to only be attached when sending a request. This removes the need for deleting the `agent` property every time we log options.

File: lib/request.js [hunk 2]
```diff
@@ -2,209 +2,215 @@
...
-            res.on('data', (chunk) => {
-                res.body += chunk;
-            });
-            res.on('end', () => {
+        delete options.authorization;
+    }
 
-                if (REDIRECT_CODES.indexOf(res.statusCode) !== -1 && redirectsCount < options.redirectMax) {
-                    options.uri = url.resolve(options.uri, res.headers.location);
-                    options.proto = options.uri.split(':')[0];
```
Comment (rerolled) [inline lib/request.js:105]: This was outright removed as it seemed to not be used anywhere.

File: lib/request.js [hunk 3]
```diff
@@ -2,209 +2,215 @@
```
Comment (rerolled) [inline lib/request.js:2]: It is recommended that you review the file on its own or with split view.

**Review by kody-abe — APPROVED** [review]
This is a great start and should provide a good foundation to move forward with the changes we are after!

**Review by tmak — APPROVED** [review]

**Review by Dus1988 — APPROVED** [review]

**Pull request merged**